src/optimizer/filter_combiner.cpp

```diff
@@ -595,40 +596,111 @@ TableFilterSet FilterCombiner::GenerateTableScanFilters(const vector<idx_t> &col
+					auto &const_val = func.children[in_val_idx]->Cast<BoundConstantExpression>();
+					auto zone_map_filter = make_uniq<ZoneMapFilter>();
+					zone_map_filter->child_filter =
+					    make_uniq<ConstantFilter>(ExpressionType::COMPARE_EQUAL, const_val.value);
+					or_filter->child_filters.push_back(std::move(zone_map_filter));
+				}
+				table_filters.PushFilter(column_index, std::move(or_filter));
 			}
+		}
+	}
 
-			sort(in_values.begin(), in_values.end());
+	for (idx_t rem_fil_idx = 0; rem_fil_idx < remaining_filters.size(); rem_fil_idx++) {
+		auto &remaining_filter = remaining_filters[rem_fil_idx];
+		if (remaining_filter->expression_class == ExpressionClass::BOUND_CONJUNCTION) {
+			auto &conj = remaining_filter->Cast<BoundConjunctionExpression>();
+			if (conj.type == ExpressionType::CONJUNCTION_OR) {
+				optional_idx column_id;
+				auto conj_filter = make_uniq<ConjunctionOrFilter>();
+				for (auto &child : conj.children) {
+					unique_ptr<ZoneMapFilter> zone_filter;
+					if (child->GetExpressionClass() != ExpressionClass::BOUND_COMPARISON) {
+						column_id.SetInvalid();
+						break;
+					}
+					optional_ptr<BoundColumnRefExpression> column_ref = nullptr;
+					optional_ptr<BoundConstantExpression> const_val = nullptr;
+					auto &comp = child->Cast<BoundComparisonExpression>();
+					if (comp.left->expression_class == ExpressionClass::BOUND_COLUMN_REF &&
+					    comp.right->expression_class == ExpressionClass::BOUND_CONSTANT) {
+						column_ref = comp.left->Cast<BoundColumnRefExpression>();
+						const_val = comp.right->Cast<BoundConstantExpression>();
+					} else if (comp.left->expression_class == ExpressionClass::BOUND_CONSTANT &&
+					           comp.right->expression_class == ExpressionClass::BOUND_COLUMN_REF) {
+						column_ref = comp.right->Cast<BoundColumnRefExpression>();
+						const_val = comp.left->Cast<BoundConstantExpression>();
+					} else {
+						// child of OR filter is not simple so we do not push the or filter down at all
+						column_id.SetInvalid();
+						break;
+					}
 
-			bool can_simplify_in_clause = true;
-			for (idx_t in_val_idx = 1; in_val_idx < in_values.size(); in_val_idx++) {
-				if (in_values[in_val_idx] - in_values[in_val_idx - 1] > 1) {
-					can_simplify_in_clause = false;
-					break;
+					if (!column_id.IsValid()) {
+						if (IsRowIdColumnId(column_ids[column_ref->binding.column_index])) {
+							break;
+						}
+						column_id = column_ids[column_ref->binding.column_index];
+					} else if (column_id.GetIndex() != column_ids[column_ref->binding.column_index]) {
+						column_id.SetInvalid();
+						break;
+					}
+
+					if (const_val->value.type().IsTemporal() ||
+					    const_val->value.type().id() == LogicalTypeId::VARCHAR) {
+						column_id.SetInvalid();
+						break;
+					}
+					auto const_filter = make_uniq<ConstantFilter>(comp.type, const_val->value);
+					zone_filter = make_uniq<ZoneMapFilter>();
```

Comment:
Instead of creating `ZoneMapFilter(x) OR ZoneMapFilter(y)`, can we make the child of a `ZoneMapFilter` the `OR` expression - i.e. `ZoneMapFilter(x OR y)`? That should allow more efficient skipping.